File: .NET/Microsoft.Recognizers.Text.Options/Extractors/IChoiceExtractorConfiguration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace Microsoft.Recognizers.Text.Options.Extractors
{
    public interface IChoiceExtractorConfiguration
    {
        IDictionary<Regex, string> MapRegexes { get; }

        Regex TokenRegex { get; }

        bool AllowPartialMatch { get; }

        int MaxDistance { get; }

        bool OnlyTopMatch { get; }
    }
}

File: Java/libraries/recognizers-text-choice/src/main/java/com/microsoft/recognizers/text/choice/extractors/IChoiceExtractorConfiguration.java
package com.microsoft.recognizers.text.choice.extractors;

import java.util.Map;
import java.util.regex.Pattern;

public interface IChoiceExtractorConfiguration {
    public Map<Pattern, String> getMapRegexes();

    public Pattern getTokenRegex();

    public boolean getAllowPartialMatch();

    public int getMaxDistance();

    public boolean getOnlyTopMatch();
}